scripts/components/api-changes-validator/api_changes_validator.test.ts

```diff
@@ -0,0 +1,102 @@
+import { after, before, describe, it } from 'node:test';
+import assert from 'node:assert';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import fsp from 'fs/promises';
+import { existsSync } from 'fs';
+import { ApiChangesValidator } from './api_changes_validator.js';
+import { execa } from 'execa';
+import { EOL } from 'os';
+
+const testResourcesPath = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  'test-resources'
+);
+const testProjectsWithBreaksPath = path.join(
+  testResourcesPath,
+  'test-projects',
+  'with-breaks'
+);
+const testProjectsWithBreaks = await fsp.readdir(testProjectsWithBreaksPath);
+const testProjectsWithoutBreaksPath = path.join(
+  testResourcesPath,
+  'test-projects',
+  'without-breaks'
+);
+const testProjectsWithoutBreaks = await fsp.readdir(
+  testProjectsWithoutBreaksPath
+);
+const workingDirectory = path.join(testResourcesPath, 'working-directory');
+
+void describe('Api changes validator', { concurrency: true }, () => {
+  before(async () => {
+    if (existsSync(workingDirectory)) {
+      await fsp.rm(workingDirectory, { recursive: true, force: true });
+    }
+    await fsp.mkdir(workingDirectory);
+  });
+
+  after(async () => {
+    await fsp.rm(workingDirectory, { recursive: true, force: true });
+  });
+
+  for (const testProject of testProjectsWithBreaks) {
+    void it(`detects break in project ${testProject}`, async () => {
+      const latestPackagePath = path.join(
+        testProjectsWithBreaksPath,
+        testProject
+      );
+      const expectedErrorMessage = (
+        await fsp.readFile(
+          path.join(latestPackagePath, 'expected-error-message.txt'),
```

Comment:
could this just be an inline string?

Reply:
this is intentional. the test is parameterized and error message is part of project definition.